DentOS_Framework/DentOsTestbed/src/dent_os_testbed/utils/test_utils/tgen_utils.py

```diff
@@ -743,3 +743,21 @@ async def tgen_utils_send_arp(device, config):
     if out[0][device.host_name]["rc"] != 0:
         device.applog.warning(f"Some arps did not reach their destination\n{out}")
     return out[0][device.host_name]["result"]
+
+
+async def tgen_utils_update_l1_config(device, ixia_ports, speed=None, autoneg=True):
+    """
+    Update ports L1 config
+    Args:
+        device (DeviceType): Ixia device
+        speed (int| None): Speed to be set on the port
+        autoneg (bool): Enable/disable autoneg on the port
+        ixia_ports (list): Ixia ports
+    """
+    device.applog.info("Changing port speed and autoneg")
+    out = await TrafficGen.update_l1_config(input_data=[{device.host_name: [
+        {"speed": speed,
+         "autoneg": autoneg,
+         "ixia_ports": ixia_ports}]}])
+    device.applog.info(out)
+    assert out[0][device.host_name]["result"] == "", f"Failed updating L1 config{out}"
```

Comment:
```suggestion
    assert out[0][device.host_name]["rc"] == 0, f"Failed updating L1 config{out}"
```

Reply:
we always return `0` and empty string  when executing ixia related code, even on fail. Better to check on `result` here

Reply:
Then don't return 0 on error. Return some error code or -1. Checking `result` here is inconsistent. In other tgen functions you always check for `rc`, so why should you check for `result` here?